url8be3f02398_vnamboo_LearnJava_tgz-pJ8-org_json_TestJ8/src/threads/currentcurrent/Buffer.java

```diff
@@ -0,0 +1,39 @@
+package threads.currentcurrent;
+
+public class Buffer {
+
+	private int data;
+	private boolean empty;
+
+	public Buffer() {
+		this.empty = true;
+	}
+
+	public synchronized void produce(int dataItem) {
+		while (!this.empty) {
+			try {
+				this.wait();
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 13:19, 27:33.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.